### DISCUSSION TITLE
Overlay like objects but evented

### DISCUSSION THREAD

**Discussion by KhorenGrig** [fabricjs/fabric.js#7531] [Q&A]
Related to this discussion [https://github.com/fabricjs/fabric.js/discussions/7480](url) , my canvas `overlayImage` is a group of Lines(like a grid). I am able to add a new Line (with button click) to that group with group.addWithUpdate(newLine) and update the existing overlayImage. Everything works well, but I want to execute events (mouseover, mousedown) only those line(s), which have been added via a button. As I understand overlayImage does not take en events, isn't it?

**Comment by asturur** [comment]
I wouldn't be against if firing an event if the current code makes a legit case unusable.
You have to consider tho that if the overlay is as large as the canvas, listenting to overlayImage-mousedown or canvas mousedown is the exact same thing.

**Comment by KhorenGrig** [comment]
@asturur Thaks for your reply. 
I set overlayImage like this `canvas.setOverlayImage(cellGridGroup, canvas.renderAll.bind(canvas))`, and try to add listener like this ` canvas.overlayImage?.on('mousedown', event => {
    console.log('[overlayImage event] ', event);
  });`. But nothing happened on *mousedown*.  If I do something wrong, let me know.  And also *canvas mousedown*  retreives *target null* for overlayImage or customGridLine.

My grid group(which is an initial overlay image) takes a part of the canvas, but the custom line (which I am adding programmatically) takes the whole canvas.

![TinyTake05-12-2021-08-04-37](https://user-images.githubusercontent.com/26667922/144754111-3b29507d-2d63-4c99-a669-a708ac2066d0.png)

*I want that the `overlayImage` grid does not take any event, but at that same time, want to add an event to the `Custom Grid Line`  object, which is a part of  `overlayImage` group*. It's possible or not?

One approach is to add that  `Custom Grid Line`  as a usual object *canvas.add(obj)*, but on this approach, I always need to bring this object *to the front* (bringToFront or moveTo), because users can add another object(s)(images, elements) that will cover that line.

**Reply by asturur** [reply]
can't you do `canvas.on('mouse:down')` and then from the event information understand if you hit the overlay image area, and execute the code?
is not clear to me if the overlay is the white square with the grid, or that if is more your design on the canvas

**Reply by KhorenGrig** [reply]
*Sorry that the provided image was not clear to you*
I can detect if *mouse:down* happened inside overlayImage area. But I need to do some extra calculations. Get event coordinates(x, y) and then compare theirs with overlayImage area coordinates in order to understand if down happened inside overlayImage. isn't it? I wanted to get a direct object(*Custom Grid Line*) on *mouse:down*.

**Reply by asturur** [reply]
the code should be simple enough, like reusing the main targeting function of fabricJS.
Do you have that grid setup in a fiddle so that we can try to hook it up?
